src/cdk/a11y/_a11y.scss

```diff
@@ -35,7 +35,8 @@
   // context. We address this by nesting the selector context under .cdk-high-contrast.
   @at-root {
     $selector-context: #{&};
-    .cdk-high-contrast-#{$target} {
+    .cdk-high-contrast-#{$target},
+     .cdk-high-contrast-#{$target} :host {
```

Comment:
I tested it out and this breaks the high contrast styles for Edge and IE for unencapsulated components. The problem is that when encapsulation is disabled, Angular leaves the `:host` in place which IE/Edge don't support and causes the entire selector to be invalidated. It can be worked around by outputting a separate selector, but it means that we generate double the amount of CSS:

```scss
.cdk-high-contrast-#{$target} {
  @content;
}

.cdk-high-contrast-#{$target} :host {
  @content;
}
```

We could also add another parameter to the mixin that allows the consumer to tell us whether the styles are encapsulated and we can output the correct styles based on it. By default it can output both.

Reply:
I'd definitely prefer to add a parameter to toggle this.

Reply:
I see- I was assuming IE would ignore a pseudo-class it didn't recognize. I pushed an update that adds a param, defaulting to emitting both styles for maximum compatibility.